icu4c/source/i18n/ucal.cpp

```diff
@@ -91,6 +91,23 @@ ucal_setDefaultTimeZone(const UChar* zoneID, UErrorCode* ec) {
     }
 }
 
+U_DRAFT int32_t U_EXPORT2
+ucal_getHostTimeZone(UChar* result, int32_t resultCapacity, UErrorCode* ec) {
+    int32_t len = 0;
+    if (ec != NULL && U_SUCCESS(*ec)) {
+        TimeZone *zone = TimeZone::detectHostTimeZone();
+        if (zone == NULL) {
+            *ec = U_MEMORY_ALLOCATION_ERROR;
```

Comment:
It's interesting - but I agree TimeZone::detectHostTimeZone() would return null only when unknown zone object cannot be created because of out of memory.

Reply:
Right, `TimeZone::getUnknown()` should give back the unknown zone if things fail.
I think maybe I might add a brief line to the API docs that this might also return NULL with U_MEMORY_ALLOCATION_ERROR in very bad cases where we can't even clone the unknown zone. What do you think?